### PR TITLE
IAM Module bugfix

### PR DESCRIPTION
**Pull request by juliedavila**
##### ISSUE TYPE

Bugfix Pull Request
##### COMPONENT NAME

iam module
##### ANSIBLE VERSION

``````
ansible 2.1.0 (iam_3346 03ec71e5af) last updated 2016/04/07 14:26:44 (GMT -400)
  lib/ansible/modules/core: (iam_3346 a662e57450) last updated 2016/04/07 16:04:46 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 7120fb4b01) last updated 2016/04/07 14:25:56 (GMT -400)
  config file = /Users/jdavila/Tech/sandbox/ansible.cfg
  configured module search path = Default w/o overrides```
``````
##### SUMMARY

Fixes all issues in #3346

Fixes issue #2821  where newly created keys were not appearing in the returned output by the module. This **DOES NOT** implement the feature request for key rotation.

There is now a `created_keys` list which will include any newly created keys where applicable.

``````
{"changed": true, "created_keys": [{"access_key_id": "FAKE3FZHKFAKE2QA", "access_key_selector": "HMAC", "create_date": "2016-04-07T20:09:11.799Z", "secret_access_key": "fake_key_alsjdflkasdfksf;lksjdfkls", "status": "Active", "user_name": "pullrequest_testing"}], "groups": null, "keys": {"FAKE3FZHKFAKE2QA": "Active"}, "user_name": "pullrequest_testing"}```
``````

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @defionscode for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/iam.py [hunk 1]
```diff
@@ -279,11 +259,31 @@ def update_user(module, iam, name, new_name, new_path, key_state, key_count, key
                 else:
                     module.fail_json(msg=error_msg)
 
+    try:
+        current_keys, status = \
```
Comment (ryansb) [inline cloud/amazon/iam.py]: Would it be possible to un-nest this? Nested list comprehensions can be hard to read. Also the use of `\` for line breaks doesn't conform to Ansible's PEP8-ish style guidelines.

**Comment by gregdek** [comment]
Thanks @defionscode for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by iainlbc** [comment]
Not the author, but this would be helpful for us :+1: 
It looks like the branch has no conflicts... ready_for_review

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/iam.py [hunk 2]
```diff
@@ -283,20 +283,12 @@ def update_user(module, iam, name, new_name, new_path, key_state, key_count, key
     if updated and new_name:
         name = new_name
     try:
-        current_keys, status = \
-            [ck['access_key_id'] for ck in
-             iam.get_all_access_keys(name).list_access_keys_result.access_key_metadata],\
-            [ck['status'] for ck in
-                iam.get_all_access_keys(name).list_access_keys_result.access_key_metadata]
+        current_keys, status = [ck['access_key_id'] for ck in iam.get_all_access_keys(name).list_access_keys_result.access_key_metadata], [ck['status'] for ck in iam.get_all_access_keys(name).list_access_keys_result.access_key_metadata]
```
Comment (ryansb) [inline cloud/amazon/iam.py:286]: Is there a reason not to break this into: 

```
current_keys = [ck['access_key_id'] for ck in iam.get_all_access_keys(name).list_access_keys_result.access_key_metadata]
status = [ck['status'] for ck in iam.get_all_access_keys(name).list_access_keys_result.access_key_metadata]
```

**Review by ryansb — COMMENTED** [review]
Looks good, just a minor style question. It looks like this changes the returns, so can you also update the docs?

**Pull request closed without merging**